### PR TITLE
refactor: move date_parser to unittest

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY

move date parser test from integration to unit test and use functional test instead of class base test.

### TESTING INSTRUCTIONS

CI

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: tests/unit_tests/utils/date_parser_tests.py [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+    elif s == "Next 5 months":
+        return datetime(2017, 4, 7)
+    elif s in ["5 days", "5 days ago"]:
+        return datetime(2016, 11, 2)
+    elif s == "2018-01-01T00:00:00":
+        return datetime(2018, 1, 1)
+    elif s == "2018-12-31T23:59:59":
+        return datetime(2018, 12, 31, 23, 59, 59)
+    else:
+        return None
```
Comment (villebro) [inline tests/unit_tests/utils/date_parser_tests.py:68]: I wonder if we could use `freezetime` from `freeze_gun` here to avoid hardcoding?

**Review by villebro — COMMENTED** [review]
Thanks for this migration! One comment about the mock, other than that LGTM

File: tests/unit_tests/utils/date_parser_tests.py (hunk 1 above)
Comment (zhaoyongjie) [reply tests/unit_tests/utils/date_parser_tests.py:68]: Thanks, Ville! Unfortunately, I'm stuck when migrate to `freeze_time`. I leave some notes for this.

```
@freeze_time(datetime(2016, 11, 7, 9, 30, 10))
def test_get_since_until() -> None:
    result: Tuple[Optional[datetime], Optional[datetime]]
    expected: Tuple[Optional[datetime], Optional[datetime]]

    # result = get_since_until()
    # expected = None, datetime(2016, 11, 7, tzinfo=pytz.utc)
    # assert result == expected

    result = get_since_until(" : now")
    # expected = None, datetime(2016, 11, 7, 9, 30, 10)
    print(result)
    # print(expected)

(superset) yongjie.zhao@:incubator-superset$ pytest -s --disable-warnings tests/unit_tests/utils/date_parser_tests.py::test_get_since_until
============================================================================= test session starts ==============================================================================
platform darwin -- Python 3.8.6, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /Users/yongjie.zhao/workspace/warehouse/superset/incubator-superset, configfile: pytest.ini
plugins: cov-2.12.1, anyio-2.2.0, celery-4.4.7, mock-3.6.1, pyfakefs-4.5.0
collected 1 item

tests/unit_tests/utils/date_parser_tests.py now
2016-11-07 17:30:10+00:00
(None, FakeDatetime(2016, 11, 7, 17, 30, 10))
```

Comment (villebro) [reply tests/unit_tests/utils/date_parser_tests.py:68]: Ok, thanks for investigating - we can leave this for a follow-up 👍

**Review by villebro — APPROVED** [review]
LGTM

**Pull request merged**